### PR TITLE
framework: improve Guest.WaitForNodesUp logs

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/4692

### CODE REVIEW

File: pkg/framework/guest.go [hunk 1]
```diff
@@ -238,16 +239,20 @@ func (g *Guest) WaitForNodesUp(numberOfNodes int) error {
...
 		if len(nodes.Items) != numberOfNodes {
-			return microerror.Maskf(waitError, "worker nodes are still not found")
+			return microerror.Maskf(waitError, "found %d k8s nodes but want %d", len(nodes.Items), numberOfNodes)
 		}
 
+		var notReadyCnt int
 		for _, n := range nodes.Items {
 			for _, c := range n.Status.Conditions {
 				if c.Type == v1.NodeReady && c.Status != v1.ConditionTrue {
-					return microerror.Maskf(waitError, "worker nodes are still not ready")
```
Comment (kopiczko) [inline pkg/framework/guest.go]: This log in particular was very misleading. There is possibility that we have more nodes than expected in the k8s api e.g. when we scale down in draining test.

**Review by rossf7 — APPROVED** [review]

File: pkg/framework/guest.go [hunk 2]
```diff
@@ -238,16 +238,20 @@ func (g *Guest) WaitForNodesUp(numberOfNodes int) error {
 		}
 
 		if len(nodes.Items) != numberOfNodes {
-			return microerror.Maskf(waitError, "worker nodes are still not found")
+			return microerror.Maskf(waitError, "found %d k8s nodes but want %d", len(nodes.Items), numberOfNodes)
 		}
 
+		var notReadyCnt int
```
Comment (xh3b4sd) [inline pkg/framework/guest.go]: Is that name on purpose? I would go with `notReadyCount`.

**Review by xh3b4sd — APPROVED** [review]

File: pkg/framework/guest.go (hunk 1 above)
Comment (xh3b4sd) [reply pkg/framework/guest.go]: I think we use something like `WaitForNodesDown` when scaling down. There is another helper for that IIRC.

File: pkg/framework/guest.go (hunk 2 above)
Comment (kopiczko) [reply pkg/framework/guest.go]: It's on purpose. It's pretty common abbreviation AFAIK. If you want provide suggestions and I change it.

**Review by MarcelMue — APPROVED** [review]
LGTM

**Pull request merged**